### ISSUE TITLE
Minio reports 0 disk size with EFS

### ISSUE THREAD

**Issue by rishka** [minio/minio#4675]
## Expected Behavior

When using the Minio Helm Chart with an EFS disk(PVC), Minio should report the correct disk size

## Current Behavior

Minio Reports 0 available bytes

## Steps to Reproduce (for bugs)

1. Start a kubernetes minio service from the helm chart using the [EFS Provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/aws/efs) for the PVC
2. Log in to the gui
3. Try and upload a file
4. Minio logs report: 
```
Drive Capacity: 8.0 EiB Free, 8.0 EiB Total
time="2017-07-13T13:34:04Z" level=error msg="Failed to create object cache/24763.pdf" cause="disk path full" source="[fs-v1.go:577:fsObjects.PutObject()]" stack="fs-v1-helpers.go:246:fsCreateFile fs-v1.go:574:fsObjects.PutObject <autogenerated>:338:(*fsObjects).PutObject web-handlers.go:506:(*webAPIHandlers).Upload web-router.go:85:(*webAPIHandlers).Upload-fm"
```

## Context

Trying to setup a solution to minimize cross-az data transfer pricing(EFS has no costs for transfer) for a CI/Build cache in a Multi-AZ kubernetes cluster

## Your Environment

* Version used (`minio version`): 
```
Version: 2017-05-05T01:14:51Z
Release-Tag: RELEASE.2017-05-05T01-14-51Z
Commit-ID: 40985cc4e3eec06b7ea82dc34c8d907fd2e7aa12
```
also happens on `minio:latest`:
```
Version: 2017-06-13T19:01:01Z
Release-Tag: RELEASE.2017-06-13T19-01-01Z
Commit-ID: 353f2d3a6e30f595a721abcd9cf1b108b041ee21
```

* Environment name and version (e.g. nginx 1.9.1): kubernetes 1.6 on AWS

**Comment by rishka** [comment]
Using `minio:edge` fails to start with `time="2017-07-13T13:58:55Z" level=error msg="Initializing object layer failed" cause="disk path full" source="[server-main.go:206:serverMain()]" `

**Comment by harshavardhana** [comment]
@Rishka can you provide more detailed instructions on how to attach EFS(PVC) with kubernetes?

**Comment by harshavardhana** [comment]
Also can you provide output of `df -i`  @Rishka on the mounted volume inside the container.. i suspect the inodes are empty or not exposed to the operating system.

**Comment by rishka** [comment]
`df -i`:
```
fs-id.efs.us-east-1.amazonaws.com://pvs/efs-minio-minio-pvc-a369e411-6776-11e7-8539-0a120f6953dc
                             0         0         0   0% /export
```
Looks like you are right.

For attaching the Persistent Volume Claim, I used [this](https://medium.com/@while1eq1/using-amazon-efs-in-a-multiaz-kubernetes-setup-57922e032776) to set up the provider

**Comment by deekoder** [comment]
@Rishka - Any chance that this started working for you?

**Comment by rishka** [comment]
Um... It did... And I didn't change anything.

Do you want me to test it with edge again

**Comment by deekoder** [comment]
Sure.

**Comment by rishka** [comment]
I still get the same error with edge, which causes the pod to not start. And switching back, latest worked right away

**Comment by nitisht** [comment]
@Rishka just to clarify, 

initially the version `latest`

```
Version: 2017-06-13T19:01:01Z
Release-Tag: RELEASE.2017-06-13T19-01-01Z
Commit-ID: 353f2d3a6e30f595a721abcd9cf1b108b041ee21
```

was not working, but now it works? Also, currently `minio/minio:edge` is not working for you?

**Comment by harshavardhana** [comment]
We don't start the server when we don't know the disk size.  We also don't start the server when we have no inodes available looks like latter is the issue here. We need to see what the disk FS type reported and ignore the check. We do this with NFS currently.

**Comment by rishka** [comment]
@NitishT Correct, I had just left `latest` running. And checked it after ~5 days. 

`minio/minio:edge` is not working because it fails on startup

@harshavardhana Is there a cli or environment variable to disable that check? I still have a pod running if you want me to check anything

**Comment by harshavardhana** [comment]
Is it possible for us to access this pod remotely like a remote ssh ?

**Comment by rishka** [comment]
No, It is a part of a production cluster. If there are commands you would like me to run, you can reach me on slack @adam_p

**Comment by nitisht** [comment]
@Rishka I will push a Minio Docker image on a private repo, could you pls test it on your setup?

**Comment by nitisht** [comment]
@Rishka please test with the image `nitisht/minio:inode`

**Comment by rishka** [comment]
Fails to start with`time="2017-07-26T13:01:05Z" level=error msg="Initializing object layer failed" cause="disk path full" source="[server-main.go:206:serverMain()]" `

**Comment by nitisht** [comment]
> Fails to start with time="2017-07-26T13:01:05Z" level=error msg="Initializing object layer failed" cause="disk path full" source="[server-main.go:206:serverMain()]"

There was an issue with the previous image, sorry for the confusion. Can you pls try `nitisht/minio:inode3` instead and let us know the server log as well.

**Comment by Rishka** [comment]
`invalid image name`

**Comment by rishka** [comment]
Still goes into a crash loop
```
time="2017-07-27T18:43:42Z" level=error msg="Initializing object layer failed" cause="disk path full" source="[server-main.go:206:serverMain()]" 
-9223372036854775808
-9223372036854775808
0
26985
NFS
```

**Comment by nitisht** [comment]
Thanks @Rishka . Let me get back to you on this

**Comment by nitisht** [comment]
@Rishka `int64` overflow caused negative values, causing a check to fail. We have fixed it, can you please try this, `nitisht/minio:inode4`

**Comment by rishka** [comment]
It starts, but trying to visit the browser fails with: 
```
Drive Capacity: 8.0 EiB Free, 8.0 EiB Total
time="2017-08-01T13:06:35Z" level=error msg="Error in reading from new connection 100.96.6.1:34746 at server 100.96.6.252:9000" cause=EOF source="[listener.go:127:github.com/minio/minio/pkg/http.(*httpListener).start.func2()]" 
```

**Comment by nitisht** [comment]
Thanks for helping us on this @Rishka , will get back to you on this error you got.

**Comment by harshavardhana** [comment]
> time="2017-08-01T13:06:35Z" level=error msg="Error in reading from new connection 100.96.6.1:34746 at server 100.96.6.252:9000" cause=EOF source="[listener.go:127:github.com/minio/minio/pkg/http.(*httpListener).start.func2()]"

This error can be ignored @Rishka - this happens due to a wrong log message.

**Comment by harshavardhana** [comment]
Do you have problems writing files @Rishka ?

**Comment by rishka** [comment]
No, it looks like it just had not started the ui yet. Everything appears to be working

**Comment by harshavardhana** [comment]
> No, it looks like it just had not started the ui yet. Everything appears to be working

Okay understood.

**Comment by nitisht** [comment]
Fixed by #4747
